pkg/runner/authenticated_token_source.go

```diff
@@ -0,0 +1,171 @@
+package runner
```

Comment:
I'm not sure this code goes in the `runner`; wouldn't it be better if it's part of `pkg/auth`?